### ISSUE TITLE
rinstall setting mode and owner when the file already exists

### ISSUE THREAD

**Issue by winduptoy** [eradman/rset#5]
```
./rinstall my-script.sh /opt/bin/my-script.sh
/opt/bin/my-script.sh
Permission denied
```
**Me:** "Oh yeah, better make it executable." So I add `-m +x` and run it again:
```
./rinstall -m +x my-script.sh /opt/bin/my-script.sh
/opt/bin/my-script.sh
Permission denied
```

Would it make sense for `rinstall` to always set permissions even if the file remains otherwise unchanged, or is there a good reason to avoid that? I'm happy to submit a PR with tests if this seems like the right thing to do.

```
if test -e "$target"; then
	[ -n "$OWNER" ] && chown $OWNER "$target"
	[ -n "$MODE" ] && chmod $MODE "$target"
fi
```

**Comment by eradman** [comment]
I can't think of a good reason to limit mode and owner changes to install/updates. I probably had in mind avoiding changes visible to rsync, but even that is only true sometimes since `ctime` (not `mtime`) is updated.

If you have time, please submit a PR for this. You can start by making a copy of the test _"No need to update a file"_. Also update `NEWS` with the feature that changed.

**Comment by winduptoy** [comment]
Having trouble getting the tests to pass in the first place. I keep running into this:

```
0.005: Run rinstall with no arguments
Traceback (most recent call last):
	3: from test_rinstall.rb:67:in `<main>'
	2: from test_rinstall.rb:39:in `try'
	1: from test_rinstall.rb:76:in `block in <main>'
test_rinstall.rb:48:in `eq': "Install a file from a remote URL to the staging area" (RuntimeError)
> chmod: cannot access '/tmp/d20210210-40420-1coorpi/test_2.txt': No such file or directory
```

I disabled the `FileUtils.remove_dir $wwwtmp` `at_exit` to see what was going on, sure enough that directory exists with what I believe to be correct permissions:

```
drwx------ 3 matt matt 4096 Feb 10 12:22 d20210210-40420-1coorpi
```
The `src` file at `x/y/test_2.txt` exists, but the `dst` does not, so I'm guessing `rinstall` failed to run...

Sorry, no Ruby experience here. :frowning_face:

**Comment by eradman** [comment]
No worries. If the tests give you to much trouble feel free to submit the change to `rinstall` and I'll build a test to exercise the change in behavior.

(Also feel free to send me an e-mail with the failing code.)

**Comment by winduptoy** [comment]
This all may be a little much ceremony for such a simple change... Sorry for complicating things!

Basically this is all I've added right at the bottom of my local copy of `rinstall` starting at line 69:

```
# Always apply permissions
if test -e "$target"; then
	[ -n "$OWNER" ] && chown $OWNER "$target"
	[ -n "$MODE" ] && chmod $MODE "$target"
fi

test -e "$target" && diff -U 2 "$target" "$source" || {
	test -e "$target" || echo "rinstall: created $target"
	cp "$source" "$target"
	[ -n "$OWNER" ] && chown $OWNER "$target"
	[ -n "$MODE" ] && chmod $MODE "$target"
	ret=0
}
exit $ret
```

**Comment by eradman** [comment]
This change appears to be very easy--there was already a test that asserted this behavior.

Implemented in change 98114e7
